packages/xo-server/src/xapi/mixins/storage.mjs

```diff
@@ -53,19 +53,29 @@ export default {
   },
 
   _getUnhealthyVdiChainLength(uuid, childrenMap, cache) {
-    let length = cache[uuid]
+    let length = cache[uuid]?.length
     if (length === undefined) {
       const children = childrenMap[uuid]
       length = children !== undefined && children.length === 1 ? 1 : 0
       try {
-        const parent = this.getObjectByUuid(uuid).sm_config['vhd-parent']
+        const parent = this.getObjectByUuid(uuid, undefined).sm_config['vhd-parent']
         if (parent !== undefined) {
           length += this._getUnhealthyVdiChainLength(parent, childrenMap, cache)
         }
       } catch (error) {
-        log.warn(`Xapi#_getUnhealthyVdiChainLength(${uuid})`, { error })
+        if (error instanceof TypeError) {
+          cache[children[0].uuid] = {
+            ...cache[children[0].uuid],
+            unknownVhdParent: uuid,
+          }
```

Comment:
`cache`  should be updated  when  `parent`  is `undefined` because `this.getObjectByUuid(uuid, undefined)` will return `undefined` if there is no object, right?